dbms/src/Debug/MockStorage.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <Debug/MockStorage.h>
+
+namespace DB::tests
+{
+void MockStorage::addTableSchema(String name, const MockColumnInfoVec & columnInfos)
+{
+    name_to_id_map[name] = MockTableIdGenerator::instance().nextTableId();
+    table_schema[getTableId(name)] = columnInfos;
+}
+
+void MockStorage::addTableData(String name, const ColumnsWithTypeAndName & columns)
+{
+    table_columns[getTableId(name)] = columns;
+}
+
+Int64 MockStorage::getTableId(String name)
```

Comment:
ditto, and the following reaad-only parameter may be written as `const String & name`